segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/BaseMetadataStore.java

```diff
@@ -209,7 +212,9 @@ public void commit(MetadataTransaction txn, boolean lazyWrite, boolean skipStore
         }
         // Step 2 : Check whether transaction is safe to commit.
         // This check needs to be atomic, with absolutely no possibility of re-entry
```

Comment:
Why? It seems like this class is concerned about cycles somehow, but how is that possible isn't there only one of these?

Reply:
The underlying TableStore stores data in table segments and any get and put on table segments might result in related pages being read/written from LTS which ends up reentering the same code.